### PR TITLE
completed array-manipulation 1-exercise-Dylan De Vei

### PR DESCRIPTION
**Pull request by dylandevei**
https://user-images.githubusercontent.com/99229316/160221961-bb85cd4f-4ebe-4abb-b473-4960382af3fc.mov

### CODE REVIEW

File: array-manipulation-1/src/initial.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* exported initial */
+// define function named initial with single parameter(array)
+// function will return all values except last in array
+// tailing values will be pushed into new array
+// first index will be taken off of the array
```
Comment (codingcodymiller) [inline array-manipulation-1/src/initial.js]: These steps sound like they correlate with `tail`, not `initial`

File: array-manipulation-1/src/reverse.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// create for loop that starts i at the end of the index and will iterate until value has reached 0
+// i will decrement by one each iteration
+// each iteration will push value to the new reverseArray
+// return value of reverse array
+
+function reverse(array) {
+  var reverseArray = [];
+  for (var i = array.length - 1; i >= 0; i--) {
+    reverseArray.push(array[i]);
+  } return reverseArray;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/reverse.js]: Please always keep `return` statements on their own line

File: array-manipulation-1/src/tail.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// i will loop until it is less than the arrays length
+// i will increment by 1
+// each iteration will push the new values after index one to tails array
+// once loop has finished the result will be returned
+
+function tail(array) {
+  var tail = [];
+  for (var i = 1; i < array.length; i++) {
+    tail.push(array[i]);
+  } return tail;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/tail.js]: Please always keep `return` statements on their own line

File: array-manipulation-1/src/initial.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// for loop to iterate through first array and push new values to new array
+
+function initial(array) {
+  var initial = [];
+  for (var i = 0; i < array.length - 1; i++) {
+    initial.push(array[i]);
+  } return initial;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/initial.js]: Please always keep `return` statements on their own line

File: array-manipulation-1/src/compact.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+// make if statement that checks if boolean value is true, if it is true then push to newArray variable
+// return newArray variable with all values that are Boolean true
+
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (Boolean(array[i]) === true) {
+      newArray.push(array[i]);
+    }
+  } return newArray;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/compact.js]: Please always keep `return` statements on their own line

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by dylandevei** [comment]
Screen Shot 2022-03-28 at 12 53 08 PM


did those requests ! Thank you for the advice :)

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**